lib/optimizely/event/entity/event_batch.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'json'
+module Optimizely
+  class EventBatch
+    attr_accessor :account_id, :project_id, :revision, :client_name, :client_version,
+                  :anonymize_ip, :enrich_decisions, :visitors
+
+    def as_json(_options = {})
+      {
+        account_id: @account_id,
+        project_id: @project_id,
+        revision: @revision,
+        client_name: @client_name,
+        client_version: @client_version,
+        anonymize_ip: @anonymize_ip,
+        enrich_decisions: @enrich_decisions,
+        visitors: @visitors
+      }
+    end
+
+    class Builder
```

Comment:
do we really need builder classes in ruby ?

Reply:
Builder is being used to handle the multi-part objects.
However can simply initialize the nested attributes as well.